### PR TITLE
Add new table for storing Genome::Qc::Config

### PR DESCRIPTION
**Pull request by susannasiebert**
``` perl
class Genome::Qc::Config {
    is => [
        'Genome::Utility::ObjectWithTimestamps',
        'Genome::Utility::ObjectWithCreatedBy',
        'Genome::Searchable',
    ],
    table_name => 'config.qc',
    id_by => [
        id => {
            is => 'Text',
            len => 64,
        }
    ],
    has => [
        name => {
            is => 'Text',
        },
        type => {
            is => 'String',
            is_optional => 1,
            valid_values => ['wgs', 'exome'],
        },
        config => {
            is => 'Text',
        }
    ],
    schema_name => 'GMSchema',
    data_source => 'Genome::DataSource::GMSchema',
    id_generator => '-uuid',
};
```

### CODE REVIEW

**Comment by susannasiebert** [comment]
The cle-integration test is failing because of two new environment variables that Nathan set back in May that aren't in the CLE snapshot. We haven't encountered this before because normal genome changes aren't tested against the CLE snapshot but this is a sqitch change. To get the test to pass we'd need a new CLE snapshot. The other option is to just merge this PR in spite of the cle-integration failure since we know that this PR is not what is making this test fail.

@ernfrid what do you think?

**Comment by acoffman** [comment]
Alright, so, I have the equivalent of the Jenkins cle-integration tests "passing" by manually changing a bunch of stuff. This is obviously very suboptimal. Also, it runs test-tracker which (as expected) determines that we don't actually need to run any tests, making this somewhat of a futile exercise in the first place. Here's a summary of what's going on:
- This will not pass on the actual `cle` snapshot because there are environment variables set that cause an immediate crash.
- `cle-staging` knows about the environment variables so we get a little further, but doesn't know about all the recent `genome-config` changes, so it breaks for other reasons
- `cle-staging` does not contain the `genome-test-env` and `genome-env` commands that are used bootstrap the tests in the first place (set up the environment, run the migrations, use the correct workflow, database, and ur versions, etc)
- Because `cle-staging` does not know about about `genome config` we cannot use `genome-test-env` from the current production snapshot - it breaks.

As such, I took the following steps to simulate the test that Jenkins is trying to run and get our successful no-op.
- In one `genome` repo, check out snapshot `genome-3598` as it is the last version of `genome-test-env` before the changes to `genome config`
- In a second `genome repo` checkout the `cle-staging` snapshot that we actually want to run tests against.
- Check out this PR into a third repo
- Modify the `genome-test-env` script from `genome-3598` to remove logic that tries to use the version of `genome-test-env` in your current repo (since it doesn't exist)
- From inside the `cle-staging` repo run the modified version of `genome-test-env` from `genome-3598` while pointing it at the checked out version of this PR:
  `/path/to/genome-3598/bin/genome-test-env -m /path/to/sqitch-pr test-tracker prove --lsf --jobs 20 -r`

This series of steps allowed the tests to actually "run" and produce the following output:

```
=> Preparing Submodules...
   ur: OK
   workflow: OK

=> Rebuilding Meta DB........... done
=> Creating test database...
=> Migrating test database...
Deploying changes to E5B6B58A201C11E5A66AA8FEA9DCD391
  + add_table_for_qc_config .. psql:deploy/add_table_for_qc_config.sql:15: NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "config_qc_pk" for table "qc"
psql:deploy/add_table_for_qc_config.sql:15: NOTICE:  CREATE TABLE / UNIQUE will create implicit index "qc_name_key" for table "qc"
ok

=> Setting Up Test Environment...

Genome Perl: 5.10.1
Genome Prove: TAP::Harness v3.17 and Perl v5.10.1
UR: /gscuser/acoffman/genome/ur/lib/UR.pm
Workflow: /gscuser/acoffman/genome/workflow/lib/Workflow.pm
***** GENOME_DEV_MODE (linus83.gsc.wustl.edu) ***** at /gscuser/acoffman/genome/lib/perl/Genome/Site/TGI/LegacyConfig.pm line 24.
Genome: /gscuser/acoffman/genome/lib/perl/Genome.pm
Test DB Snapshot: 46a7998
Test DB Name: E5B6B58A201C11E5A66AA8FEA9DCD391
Test DB Host: apipe-test-db.gsc.wustl.edu
Test DB Port: 5432

test-tracker prove --lsf --jobs 20 -r...
Files=0, Tests=0,  0 wallclock secs ( 0.00 usr +  0.00 sys =  0.00 CPU)
Result: NOTESTS
```

As expected, `test-tracker` finds no tests to run as there are no changes to any existing code. I am still unable to run full perl tests or the SomVal model test and am working on getting those to work.

File: sqitch.plan [hunk 1]
```diff
@@ -359,3 +359,5 @@ remove_params_and_inputs_from_software_results [result_software_result] 2014-12-
 
 drop_featurelist_file_id [model_feature_list] 2014-12-18T15:26:31Z Thomas B. Mooney <tmooney@genome.wustl.edu> # drop obsolete column
 @1419272844 2014-12-22T18:27:24Z Thomas B. Mooney <tmooney@genome.wustl.edu># dropped file_id column from feature lists
+
+add_table_for_qc_config [config_schema] 2015-06-24T18:54:28Z Susanna Siebert <ssiebert@linus237.gsc.wustl.edu> # This change adds a table for storing Genome::Qc::Config
```
Comment (tmooney) [inline sqitch.plan]: It looks like your e-mail address is not properly configured.

**Comment by tmooney** [comment]
+1.  I do wonder if we should start using the [UUID datatype](http://www.postgresql.org/docs/9.2/static/datatype-uuid.html) for ID columns, but we can save that for another day.

**Comment by acoffman** [comment]
Builds for both the `apipe-test-somatic-validation` and `apipe-test-somatic-validation-sv` models succeeded overnight. Their metadata is in a transient test database (otherwise we wouldn't be testing against our database change) so webviews and CLI tools won't work for viewing them, but their output directories can be found here:

`/gscmnt/gc9023/info/model_data/2889116125/build04754a62f43543e289d7aa57c62e7aa1`
`/gscmnt/gc13026/info/model_data/2889991308/build4a221c84afb6465791a4fde7374e4bbe`

This raises an interesting question: we now have untracked data on disk as a result of running these model tests. In this case I can manually delete it after we merge and verify that everything is indeed working as expected. But in the general case, how do we handle running model tests for a database change? It requires running the model tests against a test database, but then the tracking around the disk usage is lost. On the other hand, the disk usage is required so that we have output to diff against the blessed builds. Do we have a strategy to handle this that I'm unaware of?

I am almost done going through the individual failing unit tests and will post that here as well this afternoon, then I think we'll finally be ready to merge!

**Comment by acoffman** [comment]
@tmooney has referred me to JIRA issue APIPE-2999. As I figured, this is already being discussed/addressed.

**Pull request merged**